qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py

```diff
@@ -497,16 +497,74 @@ def from_operator(obj, atol=None, rtol=None):
 
     @staticmethod
-    def from_list(obj):
-        """Construct from a list [(pauli_str, coeffs)]"""
+    def from_list(obj, num_qubits=None):
+        """Construct from a list of Pauli strings and coefficients.
+
+        Each list element can be
+        * a tuple of a Pauli string and a coefficient
+        * a triple of a local, non-trivial Pauli string, an index where to apply it, and a
+          coefficient
+
+        If the Paulis are specified via the indices, the total number of qubits of the operator
+        must be specified.
+
+        For example, the 5-qubit Hamiltonian
+
+        .. math::
+
+            H = X_0 Z_3 + 2 Y_1 Y_4
+
+        can be constructed as
+
+        .. code-block:: python
+
+            # via tuples and the full Pauli string
+            op = SparsePauliOp.from_list([("XIIZI", 1), ("IYIIY", 2)])
+
+            # via triples and local Paulis with indices
+            op = SparsePauliOp.from_list([("XZ", [0, 3], 1), ("YY", [1, 4], 2)], num_qubits=5)
+
+        Args:
+            obj (Union[List[Tuple[str, complex]], List[Tuple[str, List[int], complex]]]):
+                The list of tuples (or triples) specifying the Pauli terms.
+            num_qubits (int): If the Paulis are specified as triples, the number of qubits of
+                the operator must be specified.
+
+        Returns:
+            SparsePauliOp: The SparsePauliOp representation of the Pauli terms.
+
+        Raises:
+            ValueError: If the number of qubits is required but not specified, or incompatible
+                with the indices of the Pauli terms.
+        """
         obj = list(obj)  # To convert zip or other iterable
-        num_qubits = len(obj[0][0])
-        size = len(obj)
+
+        # determine the number of qubits
+        if len(obj[0]) == 2:  # list of tuples: number of qubits is the length of the Pauli string
+            num_qubits = len(obj[0][0])
+        else:  # list of triples: the number of qubits must be specified
+            if num_qubits is None:
+                raise ValueError(
+                    "If the Paulis are specified via indices, the number of qubits must be set."
+                )
+
+        size = len(obj)  # number of Pauli terms
         coeffs = np.zeros(size, dtype=complex)
         labels = np.zeros(size, dtype=f"<U{num_qubits}")
         for i, item in enumerate(obj):
-            labels[i] = item[0]
-            coeffs[i] = item[1]
+            # tuple of Pauli string and coefficient
+            if len(item) == 2:
+                labels[i] = item[0]
+                coeffs[i] = item[1]
+            else:
+                # construct the full label based off the non-trivial Paulis and indices
+                label = ["I"] * num_qubits
+                for pauli, index in zip(item[0], item[1]):
+                    label[index] = pauli
+
+                labels[i] = "".join(label)
+                coeffs[i] = item[2]
+
```

Comment:
I didn't want to support mix the two formats, since that creates maybe not so well defined scenarios. E.g. what if you give all Pauli terms but one by the new index format: should the number of qubits be inferred from that one "full" Pauli string? I can split this into two separate functions, one for loading from index-based and one from full to make it clearer.

Reply:
You already checked the type of the first element outside the loop, so you could just put the entire loop in an `if/else` to do the two cases, rather than checking it on each loop iteration?

(Basically, I'm just all aboard the "consistency with loud failures" train)

Reply:
Yep, the for loops are adjusted 👍🏻 

I'd be still more in favor of 3-tuples for less typing but maybe @chriseclectic want's to weigh in here, if he prefers one over the other 😄